server/etcdserver/metrics.go

```diff
@@ -164,6 +164,28 @@ var (
 		Name:      "limit",
 		Help:      "The file descriptor limit.",
 	})
+	v2applySec = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "v2_apply_duration_seconds",
+		Help:      "The latency distributions of v2 apply called by backend.",
+
+		// lowest bucket start of upper bound 0.0001 sec (0.1 ms) with factor 2
+		// highest bucket start of 0.0001 sec * 2^13 == 0.8192 sec
+		Buckets: prometheus.ExponentialBuckets(0.0001, 2, 14),
+	},
+		[]string{"op"})
```

Comment:
How about having also a label (boolean) for 'success' ?  

Sometimes happens that you start to have a lot of 'very short' operations (that looks to be performance-wide good), but  in practice the operations are fact because they keep failing, so they are not doing a lot of progress ?

Reply:
Done